xrdp/xrdp_wm.c

```diff
@@ -1721,6 +1762,9 @@ callback(long id, int msg, long param1, long param2, long param3, long param4)
         case 4: /* RDP_INPUT_SCANCODE */
             rv = xrdp_wm_key(wm, param3, param1);
             break;
+        case 5: /* RDP_INPUT_UNICODE */
+            rv = xrdp_wm_key_unicode(wm, param3, param1);
+            break;
```

Comment:
It looks like the keyboard changes are independent and can be submitted separately.